src/apiGateway.ts

```diff
@@ -1,26 +1,38 @@
 import { Request } from 'express';
 import * as url from 'url';
 import { APIGatewayProxyEventV2 } from 'aws-lambda/trigger/api-gateway-proxy';
+import QueryString from 'qs';
 
 export function httpRequestToEvent(request: Request): APIGatewayProxyEventV2 {
     const headers = objectMap(request.headers, (value): string | undefined => {
         if (Array.isArray(value)) {
             return value.join(',');
         }
         if (typeof value === 'object' && value !== null) {
-            throw new Error('Unexpected query parameter value');
+            throw new Error('Unexpected header value');
         }
         return value;
     });
-    const queryStringParameters = objectMap(request.query, (value): string | undefined => {
-        if (Array.isArray(value)) {
-            return value.join(',');
-        }
-        if (typeof value === 'object' && value !== null) {
-            throw new Error('Unexpected query parameter value');
+
+    const queryStringParameters: Record<string, string> = {};
+    objectMap(
+        request.query,
+        (value: string | QueryString.ParsedQs | string[] | QueryString.ParsedQs[] | undefined, key: string) => {
+            if (Array.isArray(value)) {
+                queryStringParameters[`${key}[]`] = value.join(',');
+            } else if (typeof value === 'object' && value !== null) {
+                Object.entries(value).forEach(([k, v]) => {
+                    if (Array.isArray(v)) {
+                        queryStringParameters[`${key}[${k}][]`] = v.join(',');
+                    } else {
+                        queryStringParameters[`${key}[${k}]`] = (v ?? '').toString();
+                    }
+                });
+            } else {
+                queryStringParameters[key] = value ?? '';
+            }
```

Comment:
I admit I do not fully understand this block, but wouldn't it be possible to simply pass through the `request.query` without doing additional parsing?

Reply:
@mnapoli okay, thanks for clarifying. I trust your deep knowledge of API Gateway idiosyncrasies.